### PR TITLE
feat(29624): better errors for non-exported types

### LINKED ISSUE #29624: Better errors for non-exported types
```ts
// @filename a.ts

declare function foo(): any

declare function bar(): any;

export { bar };

// @filename b.ts

import { foo, bar } from "./a";
```

Currently we provide the following error messages:

```
Module '"./a"' has no exported member 'foo'.
Module '"./a"' has no exported member 'bar'.
```

We could instead provide the errors:

```
Module '"./a"' declares 'foo' locally, but it is not exported.
Module '"./a"' declares 'bar' locally', but it is exported as 'baz'.
```

Also using the

```
'{0}' is declared here.
```

related span.

Later on, we could provide quick fixes to respectively

* Export `foo` from ./a.
* Correct the usage of `bar` to `baz`.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #29624

### CODE REVIEW

File: tests/baselines/reference/exportLocalSpellingSuggestion.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+//// [tests/cases/conformance/es6/modules/exportLocalSpellingSuggestion.ts] ////
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/exportLocalSpellingSuggestion.js]: Technically a spelling suggestion test should reside within `compiler`, not `conformance`

File: tests/baselines/reference/exportLocalSpellingSuggestion.errors.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+==== tests/cases/compiler/a.ts (0 errors) ====
+    declare function assertNever(): void;
+    declare function foo(): void;
+    export { foo };
+    
+==== tests/cases/compiler/b.ts (1 errors) ====
+    import { assertNevar } from "./a";
+             ~~~~~~~~~~~
+!!! error TS2460: Module '"./a"' declares 'assertNever' locally, but it is exported as 'assertNevar'.
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/exportLocalSpellingSuggestion.errors.txt]: I don't think this is the correct error - `assertNever` isn't actually exported at all. This error message was meant more for the following case

```ts
// @filename: ./fileA.ts
function foo() {
}

export { foo as bar };

// @filename ./fileB.ts
import { foo } from "./fileA";
foo();
```

Comment (a-tarasyuk) [reply tests/baselines/reference/exportLocalSpellingSuggestion.errors.txt]: @DanielRosenwasser Thanks for your feedback. I used the example from the issue description. 

```ts
// @filename a.ts

declare function foo(): any
declare function bar(): any;
export { bar };

// @filename b.ts
import { foo, bar } from "./a";

// Module '"./a"' declares 'foo' locally, but it is not exported.
// Module '"./a"' declares 'bar' locally', but it is exported as 'baz'.
```

Could you provide a couple of examples for the two error messages?

Comment (DanielRosenwasser) [reply tests/baselines/reference/exportLocalSpellingSuggestion.errors.txt]: I think the example you gave is perfect, but the error message this thread is pointing to isn't correct, is it? `assertNever` isn't actually exported at all, especially not with the name `assertNevar`.

**Comment by DanielRosenwasser** [comment]
@typescript-bot pack this

**Comment by DanielRosenwasser** [comment]
@typescript-bot pack this

**Pull request merged**